core/src/main/java/org/apache/iceberg/avro/SchemaToType.java

```diff
@@ -93,8 +92,8 @@ public Type record(Schema record, List<String> names, List<Type> fieldTypes) {
       Type fieldType = fieldTypes.get(i);
       int fieldId = getId(field);
 
-      Object defaultValue = field.hasDefaultValue() && !(field.defaultVal() instanceof JsonProperties.Null) ?
-          field.defaultVal() : null;
+      Object defaultValue =
+          field.hasDefaultValue() && AvroSchemaUtil.isNonNullDefault(field.defaultVal()) ? field.defaultVal() : null;
```

Comment:
What is the value of calling `isNonNullDefault` here? It just another way to do the same check as before `!(field.defaultVal() instanceof JsonProperties.Null)`?